pkg/checks/external/main.go

```diff
@@ -1238,21 +1242,33 @@ func (ext *Checker) Shutdown() error {
 		ext.shutdownCTXFunc()
 	}
 
+	doneChan := make(chan error, 1)
+
 	// make a context to track pod removal and cleanup
 	ctx, _ := context.WithTimeout(context.Background(), ext.Timeout())
 
-	// make sure the pod is gone before we shutdown
-	err := ext.waitForShutdown(ctx)
-	if err != nil {
-		ext.log("Error waiting for pod removal during shutdown:", err)
-		return err
-	}
-
-	// wait for all background checkers and workers to finish before the check is fully "shutdown"
-	ext.log("Waiting for background workers to cleanup...")
-	ext.wg.Wait()
+	go func() {
+		log.Debugln("Waiting for pod", ext.podName(), "to shutdown")
+		doneChan <- ext.waitForShutdown(ctx)
```

Comment:
Why not just put this directly in the select below on line 1256?  It seems like all this go routine adds is a couple logs.  Can we remove `doneChan` completely here?

Reply:
yupp but i'll have to modify `waitForShutdown` to return a `chan error`